normalize/diff.py

```diff
@@ -660,6 +664,20 @@ def compare_collection_iter(propval_a, propval_b, fs_a=None, fs_b=None,
             rev_key[(pk, seen[pk])] = k
             seen[pk] += 1
 
+    if options.recurse:
+        # we can be sure that both records have these keys
```

Comment:
OK I see.  looking over the code change, it seems that this makes the code descend matching by key/index instead of value.  The code it skips treats collections as bags of values, with the key being a relatively non-interesting "fact" about the member in the collection.  ie, for most of the code, the key didn't matter as much as the "primary key" field (or fuzzy content match) when it came to matching items for comparison.  

So this option might better be called `match_keys` or `match_indices`. Or maybe `use_indices_as_keys` or something shorter than that.